### PR TITLE
Added Basic Server and Client using DatagramSocket.

### PR DESCRIPTION
**Pull request by bananapizzza**
Added test case for the new classes.

### CODE REVIEW

File: .idea/libraries/junit_junit_4_11.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<component name="libraryTable">
```
Comment (xaveng) [inline .idea/libraries/junit_junit_4_11.xml]: Let's add .gitignore and ignore .idea directory and amend this commit.

Comment (xaveng) [reply .idea/libraries/junit_junit_4_11.xml]: out directory as well.

File: src/Client.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import java.io.IOException;
+import java.net.*;
+
+public class Client {
+    final int PORT = 9876;
+    DatagramSocket clientSocket;
+
+    public Client() {
+        System.out.println("Client()");
```
Comment (xaveng) [inline src/Client.java]: Let's remove debugging printouts.

File: src/Client.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    public String sendMessageToServer(String msg) {
+        System.out.println("sendMessageToServer()");
+
+        byte[] buf;
+        String receivedSentence = null;
+        InetAddress IPAddress = null;
+        try {
+            IPAddress = InetAddress.getByName("localhost");
```
Comment (xaveng) [inline src/Client.java]: Let's pull "localhost" to a final field as well.

File: src/Client.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+        try {
+            clientSocket = new DatagramSocket();
+        } catch (SocketException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public String sendMessageToServer(String msg) {
+        System.out.println("sendMessageToServer()");
```
Comment (xaveng) [inline src/Client.java]: Same here.

File: src/Server.java [hunk 5]
```diff
@@ -1,2 +1,53 @@
...
+
+            DatagramPacket sendPacket = new DatagramPacket(buf, buf.length, IPAddress, port);
+            try {
+                serverSocket.send(sendPacket);
+                System.out.println("packet sent to client");
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+
+            if (receivedSentence.equals("end")) {
```
Comment (xaveng) [inline src/Server.java]: Let's pull "end" to public static final field so client can also use this field.

File: udp_socket.iml [hunk 6]
```diff
@@ -4,8 +4,10 @@
     <exclude-output />
```
Comment (xaveng) [inline udp_socket.iml]: Add *.iml to .gitignore file as well.

File: .idea/libraries/junit_junit_4_11.xml (hunk 1 above)
Comment (bananapizzza) [reply .idea/libraries/junit_junit_4_11.xml]: okay

File: src/Client.java (hunk 2 above)
Comment (bananapizzza) [reply src/Client.java]: okay

File: src/Client.java (hunk 3 above)
Comment (bananapizzza) [reply src/Client.java]: okay

File: src/Client.java (hunk 4 above)
Comment (bananapizzza) [reply src/Client.java]: okay

File: src/Server.java (hunk 5 above)
Comment (bananapizzza) [reply src/Server.java]: okay

**Pull request merged**